extensions/replication/ReplicationConfigValidator.js

```diff
@@ -36,22 +51,22 @@ const joiSchema = joi.object({
             }).when('type', { is: 'role', then: joi.required() }),
         }).required(),
     },
-    destination: {
+    destination: joi.object({
         transport: transportJoi,
-        auth: joi.object({
-            type: joi.alternatives().try('account', 'role', 'service')
-                .required(),
-            account: joi.string()
-                .when('type', { is: 'account', then: joi.required() }),
-            vault: joi.object({
-                host: joi.string().optional(),
-                port: joi.number().greater(0).optional(),
-                adminPort: joi.number().greater(0).optional(),
-                adminCredentialsFile: joi.string().optional(),
-            }),
-        }).required(),
+        auth: destinationAuthJoi.optional(),
+        sites: joi.object().pattern(
```

Comment:
* there is an `authJoi` type in configItems.joi.js, which holds both transport and auth: can't we use it (or extend it) instead of introducting a new destinationAuthJoi type?

* once we have a type with transport+auth, maybe `destination` can extend this type to add bootstrap? (but maybe this creates lots of override, to make things optional/alternative)